api/src/test/java/org/openmrs/validator/VisitValidatorTest.java

```diff
@@ -448,4 +455,78 @@ public void validate_shouldFailValidationIfFieldLengthsAreNotCorrect() {
 		new VisitValidator().validate(visit, errors);
 		assertEquals(true, errors.hasFieldErrors("voidReason"));
 	}
+	
+	@Test
+	public void validate_shouldFailValidationIfVisitDateBeforeBirthDate() {
```

Comment:
we should have test cases

* `validate_shouldPassValidationIfVisitDateOnBirthDate` - with the same patient setup as here and visit calendar.set(1974, 4, 8)
* `validate_shouldPassValidationIfVisitDateAfterBirthDate` - with the same patient setup as here and visit calendar.set(1974, 4, 9)

Reply:
Added below